services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java

```diff
@@ -748,6 +759,38 @@ protected Response buildLogoutResponse(UserSessionModel userSession, String logo
     public Response backchannelLogout(UserSessionModel userSession, AuthenticatedClientSessionModel clientSession) {
         ClientModel client = clientSession.getClient();
         SamlClient samlClient = new SamlClient(client);
+
+        // real backchannel logout if SOAP binding is supported (#9548)
+        String soapLogoutUrl = getLogoutServiceUrl(session, client, SAML_SOAP_BINDING, true);
+        if (soapLogoutUrl != null) {
+            try {
+                LogoutRequestType logoutRequest = createLogoutRequest(soapLogoutUrl, clientSession, client);
+                Document samlLogoutRequest = createBindingBuilder(samlClient, false).postBinding(SAML2Request.convert(logoutRequest)).getDocument();
+                SOAPMessage soapResponse = Soap.createMessage().addToBody(samlLogoutRequest).call(soapLogoutUrl);
+                Document logoutResponse = Soap.extractSoapMessage(soapResponse);
+                SAMLDocumentHolder samlDoc = SAML2Response.getSAML2ObjectFromDocument(logoutResponse);
+                if (!(samlDoc.getSamlObject() instanceof StatusResponseType)) {
+                    throw new RuntimeException("Logout response received from client is not an StatusResponseType");
+                }
+                StatusResponseType logoutStatus = (StatusResponseType) samlDoc.getSamlObject();
+                if (!Objects.equals(logoutStatus.getStatus().getStatusCode().getValue().toString(), JBossSAMLURIConstants.STATUS_SUCCESS.get())) {
+                    logger.warnf("saml logout reported as not successful for client: %s; status=%s", client.getClientId(),
+                            logoutStatus.getStatus().getStatusCode().getValue());
+                }
+            } catch (SOAPException e) {
+                logger.warnf(e, "Logout failed for client %s", client.getClientId());
+                return Response.ok().build();
+            } catch (Exception e) {
+                logger.warn("failed to execute saml soap logout", e);
+                return Response.serverError().build();
```

Comment:
Why a error response whereas the previous one is ok ?
Why do we return a response ?

Reply:
We return a response because of the method signature. The callers are either ignoring the returned value or they record the status of the response for each client.